narwhals/functions.py

```diff
@@ -1040,6 +1040,42 @@ def func(plx: Any) -> Any:
     return Expr(func, ExprMetadata.selector())
 
 
+def exclude(*names: str | Iterable[str]) -> Expr:
+    """Creates an expression that excludes columns by their name(s).
+
+    Arguments:
+        names: Name(s) of the columns to exclude.
+
+    Returns:
+        A new expression.
+
+    Examples:
+        >>> import polars as pl
+        >>> import narwhals as nw
+        >>>
+        >>> df_native = pl.DataFrame({"a": [1, 2], "b": [3, 4], "c": ["x", "z"]})
+        >>> nw.from_native(df_native).select(nw.exclude("c", "a"))
+        ┌──────────────────┐
+        |Narwhals DataFrame|
+        |------------------|
+        |  shape: (2, 1)   |
+        |  ┌─────┐         |
+        |  │ b   │         |
+        |  │ --- │         |
+        |  │ i64 │         |
+        |  ╞═════╡         |
+        |  │ 3   │         |
+        |  │ 4   │         |
+        |  └─────┘         |
+        └──────────────────┘
+    """
+
+    def func(plx: Any) -> Any:
+        return plx.exclude(*flatten(names))
```

Comment:
I think you'd be able to simplify each backend by changing this to:

```suggestion
        return plx.exclude(set(flatten(names))
```

Then changing what they accept to be a single (non-variadic) argument:
```py
column_names: set[str]
```

Reply:
Could alternatively annotate it as:
```py
column_names: Container[str]
```

Since only `__contains__` is used

Reply:
Resolved in (https://github.com/narwhals-dev/narwhals/pull/2122/commits/07deb3a365d11a4ea1ee09313538a6af9aa3842f)